### PR TITLE
Added the solution of problem 0102, with preferable changes

### PR DESCRIPTION
**Pull request by Srijita-Mandal**
## Change Summary

Added the solution of the leetcode problem 102 with preferable changes

## Checklist

**If you haven't fulfilled the below requirements, your PR won't be reviewed.**

### General 

- [x] Make sure you've read [CONTRIBUTING.md](https://github.com/wingkwong/leetcode-the-hard-way/blob/main/CONTRIBUTING.md) 
- [x] Make sure you've followed the template format
- [x] Make sure you've started the app locally and verified all the content and all links (if applicable) must be accessible correctly
- [x] Make sure you've applied LaTex for variables, formulas and time / space complexity instead of using backticks
- [x] Make sure you've formatted your code well (Code Block is rendered using `<Tabs/>`)
      Below is a C++ example. You can include multiple langs using `<TabItem/>` if necessary
      
````
<Tabs>

<TabItem value="cpp" label="C++">
<SolutionAuthor name="@YOUR_ALIAS"/>

```cpp
// Your code goes here
// Prefer K&R Coding style
```

</TabItem>

</Tabs>
````
- [x] Make sure you've explained your code well (write comments above each line - not inline)
- [x] Make sure you've written your explanation well and it is easy to understand for beginners
- [x] Make sure you've included Time & Space Complexity Analysis
- [x] Make sure you've included a meaningful approach name for your solution. Don't leave it blank. e.g. `## Approach 1: Two Pointers`.

### For Tutorial Articles

- [ ] Make sure you've explained your topic using 1 - 3 LC problems rather than using external problems. Solutions must be taken with explanations as well.
- [ ] Make sure you've provided suggested problems at the end using the given format. See [here](https://raw.githubusercontent.com/wingkwong/leetcode-the-hard-way/main/tutorials/math/number-theory/binary-exponentiation.md) as an example. If the target solution is not available, leave `solutionLink` blank.
- [ ] Make sure you've given credits / references if you use external resources. For an image, give credit under it. Otherwise, add a new section called References at the end (after Suggested Problems).

### CODE REVIEW

File: solutions/0100-0199/0102-binary-tree-level-order-traversal-medium.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+**Space Complexity:**  $O(N)$
+
+
+<Tabs>
+<TabItem value="cpp" label="C++">
+<SolutionAuthor name="@Srijita-Mandal"/>
+
+```cpp
+class Solution {
+public:
```
Comment (wingkwong) [inline solutions/0100-0199/0102-binary-tree-level-order-traversal-medium.md:63]: Adopt in K&R coding style. 

Ref: https://gist.github.com/jesseschalken/0f47a2b5a738ced9c845#why-kr

File: solutions/0100-0199/0102-binary-tree-level-order-traversal-medium.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+## Approach 1: Using Queue Data Structure
+
+- Take one 2D vector $ans$ to return the final tree-traversal, and another vector $v$ to store seperate levels.
+- Take a queue, store $root$ node inside it and a _NULL_ node for seperating levels.
+- Until the queue is not empty, pop one by one nodes from queue. if the node is _NULL_ and queue is not empty, push the _NULL_ again inside the queue. Push $v$ into $ans$ as one level is completed and clear $v$.
+- If the node is not null, push the value into $v$ and push its left and right nodes into queue (if not null).
+
+**Time Complexity:**  $O(N)$
+
+**Space Complexity:**  $O(N)$
```
Comment (wingkwong) [inline solutions/0100-0199/0102-binary-tree-level-order-traversal-medium.md:54]: why O(N)? explain it to beginners.

**Review by wingkwong — CHANGES_REQUESTED** [review]
Hope it's final round

**Comment by Srijita-Mandal** [comment]
Committed the suggested changes.

**Comment by Srijita-Mandal** [comment]
@wingkwong Are the changes okay now? Please review the PR.

**Comment by wingkwong** [comment]
To save time, I've cleaned it up.

**Pull request merged**